stdlib/public/core/Flatten.swift.gyb

```diff
@@ -313,40 +270,50 @@ public struct ${Collection}<
   /// reachable from `startIndex` by zero or more applications of
   /// `successor()`.
   public var endIndex: Index {
-    return ${Index}(_endIndexOfFlattened: _base)
+    return ${Index}(_base.endIndex, nil)
   }
```

Comment:
Ah, I wasn't thinking about storage. That makes sense, thanks. In that case I think the current implementation is the right one.